persistence-api/src/main/java/io/stargate/db/query/SchemaKey.java

```diff
@@ -67,7 +67,7 @@ public boolean equals(Object o) {
     if (this == o) {
       return true;
     }
-    if (o == null || getClass() != o.getClass()) {
+    if (!(o instanceof SchemaKey)) {
```

Comment:
In this particular case this change looks questionable to me. We have two sub-classes: `PartitionKey` and `PrimaryKey`, assuming the table does not have clustring columns those objects may be mixed up... Do we want to allow equality between a `PartitionKey` instance and a `PrimaryKey` instance in principle? @pcmanus WDYT?

Reply:
My bad, I didn't notice there were subclasses here.

My guess is that it doesn't matter because the two implementations are probably never compared with each other. But I can revert the changes to keep it safe.

Reply:
I do would prefer reverting the change and using a suppression (and a short comment). For all that I tend to agree with the argument that `instanceof` is a better "default", I don't think it's an absolute, and that's imo an example.

And yes, pretty sure we never compare `PrimaryKey` with `PartitionKey` now, but that could create confusion some day, and the goal of Error Prone is to help us get code that is less prone to errors, not more.